### ISSUE TITLE
Panic during checking

### ISSUE THREAD

**Issue by RazrFalcon** [anderejd/cargo-geiger#19]
I'm trying to build [resvg/tools/rendersvg](https://github.com/RazrFalcon/resvg/tree/master/tools/rendersvg):
```
% RUST_BACKTRACE=1 cargo geiger
    Updating registry `https://github.com/rust-lang/crates.io-index`
    Updating git repository `https://github.com/RazrFalcon/resvg-qt`
    Updating git repository `https://github.com/RazrFalcon/usvg`                
    Updating git repository `https://github.com/RazrFalcon/svgdom`              
    Updating git repository `https://github.com/RazrFalcon/svgtypes`            
    Updating git repository `https://github.com/RazrFalcon/xmlparser`           
    Updating registry `https://github.com/rust-lang/crates.io-index`            
 Downloading fern v0.5.6
 Downloading libflate v0.1.16                                                   
   Compiling num-traits v0.2.5                                                  
    Checking cfg-if v0.1.4
   Compiling build_const v0.2.1
    Checking siphasher v0.2.2
    Checking nodrop v0.1.12
    Checking adler32 v1.0.3
    Checking byteorder v1.2.3
    Checking simplecss v0.1.0
    Checking slab v0.4.0
    Checking safemem v0.2.0
    Checking unicode-segmentation v1.2.1
    Checking rctree v0.2.1
    Checking unicode-width v0.1.5
    Checking libc v0.2.42
    Checking log v0.4.3
    Checking arrayvec v0.4.7
    Checking phf_shared v0.7.22
   Compiling crc v1.8.1
    Checking base64 v0.9.2
    Checking getopts v0.2.18
    Checking xmlparser v0.4.1 (https://github.com/RazrFalcon/xmlparser?rev=52a57d9#52a57d99)
    Checking fern v0.5.6
    Checking phf v0.7.22
    Checking time v0.1.40
    Checking libflate v0.1.16
    Checking num-traits v0.1.43
    Checking float-cmp v0.4.0
    Checking euclid v0.17.3
    Checking svgtypes v0.1.1 (https://github.com/RazrFalcon/svgtypes?rev=a794d41#a794d416)
    Checking svgdom v0.13.0 (https://github.com/RazrFalcon/svgdom?rev=968db94#968db94b)
    Checking lyon_geom v0.10.1
    Checking usvg v0.2.0 (https://github.com/RazrFalcon/usvg?rev=803c4f7#803c4f7b)
    Checking resvg v0.3.0 (file:///media/data/Programming/Sources/resvg)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:945:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:463
   5: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:350
   6: rust_begin_unwind
             at libstd/panicking.rs:328
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   8: core::result::unwrap_failed
   9: <core::iter::Map<I, F> as core::iter::iterator::Iterator>::fold
  10: <cargo_geiger::CustomExecutor as cargo::core::compiler::Executor>::exec
  11: <F as cargo::core::compiler::job::FnBox<A, R>>::call_box
  12: <F as cargo::core::compiler::job::FnBox<A, R>>::call_box
  13: <F as cargo::core::compiler::job::FnBox<A, R>>::call_box
  14: cargo::core::compiler::job::Job::run
  15: <F as crossbeam::FnBox>::call_box
  16: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  17: <F as alloc::boxed::FnBox<A>>::call_box
  18: std::sys_common::thread::start_thread
             at /checkout/src/liballoc/boxed.rs:648
             at libstd/sys_common/thread.rs:24
  19: std::sys::unix::thread::Thread::new::thread_start
             at libstd/sys/unix/thread.rs:90
  20: <unknown>
  21: clone
^C
```

**Comment by anderejd** [comment]
Thanks for the report! I will look at this. The whole thing is riddled with .unwrap and friends and I'm pretty sure this can happen if something in the CustomExecutor panics. I'll replace all .unwrap and .expect with Result based error handling to begin with and take a closer look at this issue once that's in place.

**Comment by anderejd** [comment]
Fixed by #23
